cel/options.go

```diff
@@ -387,6 +391,28 @@ func EvalOptions(opts ...EvalOption) ProgramOption {
 	}
 }
 
+// CostTracking enables cost tracking and registers a ActualCostEstimator that can optionally provide a runtime cost estimate for any function calls.
+// This enables runtime costs to be assigned to calls function library extensions.
+func CostTracking(costEstimator interpreter.ActualCostEstimator) ProgramOption {
+	return func(p *prog) (*prog, error) {
+		p.callCostEstimator = costEstimator
+		p.evalOpts |= OptTrackCost
+		return p, nil
+	}
+}
+
+// CostLimit enables cost tracking and sets configures program evaluation to exit early with a
+// "runtime cost limit exceeded" error if the runtime cost exceeds the costLimit.
+// The CostLimit is a metric that corresponds to the number and estimated expense of operations
+// performed while evaluating an expression. It is indicative of CPU usage, not memory usage.
+func CostLimit(costLimit uint64) ProgramOption {
+	return func(p *prog) (*prog, error) {
+		p.costLimit = &costLimit
+		p.evalOpts |= OptTrackCost
```

Comment:
Is it an error to provide a CostLimit and not enable `CostTracking`? If so, I'd make sure that cost tracking is only enabled in one of the two functions and make sure that this option will have no effect unless CostTracking is also enabled.

Reply:
Any combination of CostLimit, CostTracking and OptTrackCost is valid and allowed. Just OptTrackCost will track cost using the default cost calculations, CostTracking allows the cost calculations to extended, and CostLimit enables a limit. Both CostTracking and CostLimit enable OptTrackCost so that they behave as expected.